cloudbridge/cloud/base/resources.py

```diff
@@ -356,12 +356,16 @@ class BasePageableObjectMixin(PageableObjectMixin):
     """
 
     def __iter__(self):
-        result_list = self.list()
+        for result in self.iter():
+            yield result
+
+    def iter(self, **kwargs):
+        result_list = self.list(**kwargs)
         if result_list.supports_server_paging:
             for result in result_list:
                 yield result
             while result_list.is_truncated:
-                result_list = self.list(marker=result_list.marker)
+                result_list = self.list(marker=result_list.marker, **kwargs)
```

Comment:
There's a bit of an issue here - at an interface level, the list method does not accept keyword args, it only accepts limit and marker. The list method does not support filtering of any kind. In contrast, the find() method accepts kwargs (and probably ought to have been renamed to filter instead). We think that perhaps the thing to do here is to move the iter() implementation into the GCE base class instead, and call self.find() instead of self.list(). At some future point, we can revisit the list method and consider whether to change the interface contract, but right now, it would require sweeping interface changes.

Reply:
@chiniforooshan What are your thoughts on this?

Reply:
The list() method does support filtering, even at an interface level, for some services, e.g. SubnetService. So, IMHO, this is the right place for iter(). WDYT?

Also, the find() methods' return types seem to be inconsistent at the interface level; sometimes they return lists, sometimes they return a single object.

Reply:
Hmm.. you're right. That is a bit inconsistent. The list method should ideally have a fixed signature without filtering capabilities. The find method is for filtering and should return a list, never a single record. The plan was to eventually have the find method superseded by chainable filters, mimicking what django, boto3 etc. do. I'm ok with leaving this as is for now, this probably needs to be revised across the whole of cloudbridge.